### ISSUE TITLE
Rule MA0038 (Make method static) should not apply to partial methods

### ISSUE THREAD

**Issue by louis-z** [meziantou/Meziantou.Analyzer#192]
**Context**

Rule identifier: _MA0038_
Target Framework: _.NET Standard 2.0_
Language version: _8.0_

**Describe the bug**

We have code that contains auto-generated `partial` method definitions and developer-written implementations for them. The MA0038 rule reports that some of those implementations could be made static, but since we don't control the generated code (and therefore cannot also make the partial method definitions static), this is not possible.
 
**To Reproduce**

In a file, say `Class1.Generated.cs`, we have the partial method definition:

````c#
namespace ClassLibrary1
{
    public partial class Class1
    {
        partial void DoSomething();
    }
}
````

In another file, say `Class1.cs`, we have the partial method implementation:

````c#
namespace ClassLibrary1
{
    public partial class Class1
    {
        partial void DoSomething() { }
    }
}
````
